nixos/modules/profiles/base.nix

```
@@ -51,7 +51,9 @@
   ];
 
   # Include support for various filesystems.
-  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "zfs" "ntfs" "cifs" ];
+  boot.supportedFilesystems =
+    [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "ntfs" "cifs" ] ++
+    lib.optional (lib.meta.availableOn pkgs.stdenv.hostPlatform pkgs.zfs) "zfs";
```

```suggestion
    lib.optional (lib.meta.availableOn pkgs.stdenv.hostPlatform config.boot.zfs.package) "zfs";
```

Just in case e.g. they added a new platform in unstable but it hadn't made it to stable yet.